cognite_toolkit/_cdf_tk/loaders/_resource_loaders/transformation_loaders.py

```diff
@@ -249,6 +259,50 @@ def load_resource_file(
                     hash_str = f"{self._hash_key}: {auth_hash}"
                     if not item["query"].startswith(self._hash_key):
                         item["query"] = f"{hash_str}\n{item['query']}"
+
+            if "sourceOidcCredentials" in item:
+                HighSeverityWarning(
+                    "The property 'sourceOidcCredentials' is deprecated. Use 'authentication.read' instead."
+                ).print_warning(console=self.console)
+                item.pop("sourceOidcCredentials")
+
+            if "destinationOidcCredentials" in item:
+                HighSeverityWarning(
+                    "The property 'destinationOidcCredentials' is deprecated. Use 'authentication.write' instead."
+                ).print_warning(console=self.console)
+                item.pop("destinationOidcCredentials")
+
+            if "sourceNonce" in item:
+                HighSeverityWarning(
+                    "The property 'sourceNonce' is not used by Toolkit. Use 'authentication.read' instead,"
+                    "then Toolkit will dynamically set the nonce."
+                ).print_warning(console=self.console)
+                item.pop("sourceNonce")
+
+            if "destinationNonce" in item:
+                HighSeverityWarning(
+                    "The property 'destinationNonce' is not used by Toolkit. Use 'authentication.write' instead,"
+                    "then Toolkit will dynamically set the nonce."
+                ).print_warning(console=self.console)
+                item.pop("destinationNonce")
+            auth = item.pop("authentication", None)
+            if isinstance(auth, dict):
+                self._authentication_by_id_operation[(external_id, "read")] = read_auth(
+                    auth["read"] if "read" in auth else auth,
+                    self.client.config,
+                    external_id,
+                    "transformation",
+                    allow_oidc=True,
+                    console=self.console,
+                )
+                self._authentication_by_id_operation[(external_id, "write")] = read_auth(
+                    auth["write"] if "write" in auth else auth,
```

Comment:
this will probably misbehave if auth.read exists but no auth.write, so a more detailed check, if any, may be needed